### PR TITLE
Added support for RolloutSpec Create, update, patch, get, get all and…

### PR DESCRIPTION
**Pull request by janetlinkiruba**
… Delete API's

Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : No
- I have compiled the code successfully my laptop : No
- I have written examples to demonstrates the usage of the newly implemented APIs : No
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java [hunk 1]
```diff
@@ -169,4 +170,124 @@ public Boolean deleteStrategy(String strategyName) {
...
+            HttpError       httpException  = httpExceptions.get(0);
+            LOGGER.error(String.format("Error encountered while attempting to get Ocean CD RolloutSpec. Code: %s. Message: %s.",
+                    httpException.getCode(), httpException.getMessage()));
+            throw new SpotinstHttpException(httpException.getMessage());
+        }
+
+        return retVal;
+    }
+
+    public Boolean updateRolloutSpec(RolloutSpec rolloutSpecReq, String strategyName) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java:233]: Change the input variable to specName

File: src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java [hunk 2]
```diff
@@ -169,4 +170,124 @@ public Boolean deleteStrategy(String strategyName) {
...
+            HttpError       httpException  = httpExceptions.get(0);
+            LOGGER.error(
+                    String.format("Error encountered while attempting to update Ocean CD RolloutSpec. Code: %s. Message: %s.",
+                            httpException.getCode(), httpException.getMessage()));
+            throw new SpotinstHttpException(httpException.getMessage());
+        }
+        return retVal;
+    }
+
+    public Boolean patchRolloutSpec(RolloutSpec rolloutSpecReq, String strategyName) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java:253]: Change variable name to rolloutSpecName

File: src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiIstioVirtualServices.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiIstioVirtualServices implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiIstioVirtualServices.java:16]: Should be renamed to ApiIstioVirtualService

File: src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiNginxAdditionalIngressAnnotations.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiNginxAdditionalIngressAnnotations implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiNginxAdditionalIngressAnnotations.java:15]: From where the parameters of this object are taken?

File: src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiRolloutSpecStrategy.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiRolloutSpecStrategy implements IPartialUpdateEntity {
+    @JsonIgnore
+    private Set<String>                               isSet;
+    private String                                    name;
+    private List<ApiBackgroundVerificationArgs>          args;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiRolloutSpecStrategy.java:21]: Should it be ApiRolloutSpecStrategyArgs

File: src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/IstioVirtualServices.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class IstioVirtualServices {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/IstioVirtualServices.java:13]: Same comment as API

File: src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/RolloutSpecStrategy.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class RolloutSpecStrategy {
+    @JsonIgnore
+    private Set<String>                               isSet;
+    private String                                    name;
+    private List<BackgroundVerificationArgs>          args;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/RolloutSpecStrategy.java:17]: Should it be RolloutSpecArgs?

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java [hunk 8]
```diff
@@ -196,6 +196,400 @@ public static ApiStepsSetCanaryScale toDal(StepsSetCanaryScale src) {
         return apiStepsSetCanaryScale;
     }
 
+    public static ApiRolloutSpec toDal(RolloutSpec src) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:199]: Can we have separate convertor class for each type - strategy, RolloutSpec etc.?

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
Review completed. Please review the comments and make appropriate fixes.

File: src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java (hunk 1 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java:233]: updated

File: src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java (hunk 2 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/OceanCDClient.java:253]: updated

File: src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiIstioVirtualServices.java (hunk 3 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiIstioVirtualServices.java:16]: updaeted

File: src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiRolloutSpecStrategy.java (hunk 5 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/oceanCD/ApiRolloutSpecStrategy.java:21]: I am re-using the same class used in Strategy

File: src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/RolloutSpecStrategy.java (hunk 7 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/RolloutSpecStrategy.java:17]: I am re-using the same class used in Strategy

File: src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/IstioVirtualServices.java (hunk 6 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/oceanCD/IstioVirtualServices.java:13]: fixed

File: src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java (hunk 8 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/converters/oceanCD/OceanCDConverter.java:199]: separated the converter class and tested

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**